labman/gui/handlers/process_handlers/quantification_process.py

```diff
@@ -37,9 +39,45 @@ def post(self):
             pc = plate.plate_configuration
             concentrations = QuantificationProcess.parse(
                 file_content, rows=pc.num_rows, cols=pc.num_columns)
+
+            names = np.empty_like(plate.layout, dtype='object')
+            blanks = np.zeros_like(plate.layout, dtype=bool)
+
+            # fetch the sample names and whether or not the samples are blanks
+            # by default these are set to be None and False.
+            for i, full_row in enumerate(plate.layout):
+                for j, well in enumerate(full_row):
+
+                    # some wells have no compositions at all so skip those
+                    if well is None:
+                        continue
+                    comp = well.composition
+
+                    # cache the sample compositions to avoid extra intermediate
+                    # queries
+                    if isinstance(comp, LibraryPrep16SComposition):
```

Comment:
I think you also need two more types here: `GDNAComposition` and `CompressedGDNAComposition`, since those two types can also be quantified.
To get the sample composition from a `GDNAComposition`:
```python
comp.sample_composition
```
from a `CompressedGDNAComposition`:
```python
comp.gdna_composition.sample_composition
```